### ISSUE TITLE
Export LayerNorm to ONNX lead to wrong output shape info.

### ISSUE THREAD

**Issue by grimoire** [pytorch/pytorch#84563]
### 🐛 Describe the bug

Export layer norm to ONNX (without dynamic axis) would give a ONNX with wrong output shape.

```python
model = torch.nn.LayerNorm(256)
x = torch.rand(950, 1, 256)
torch.onnx.export(
    model,
    x,
    'tmp.onnx',
    input_names=['input'],
    output_names=['output'],
    opset_version=11)
```

If we print the graph after `torch._C._jit_pass_onnx`:

```
graph(%input : Float(950, 1, 256, strides=[256, 256, 1], requires_grad=0, device=cpu), 
      %layer.weight : Float(256, strides=[1], requires_grad=1, device=cpu), 
      %layer.bias : Float(256, strides=[1], requires_grad=1, device=cpu)): 
  %6 : Double(device=cpu) = onnx::Constant[value={2}]()
  %7 : Double(device=cpu) = onnx::Constant[value={1e-05}]() 
  %8 : Float(950, 1, device=cpu) = onnx::ReduceMean[axes=[-1]](%input) 
  %9 : Float(950, 1, 256, strides=[243200, 256, 1], device=cpu) = onnx::Sub(%input, %8) 
  %10 : Float(950, 950, 256, strides=[256, 256, 1], device=cpu) = onnx::Pow(%9, %6)
  %11 : Float(950, 1, device=cpu) = onnx::ReduceMean[axes=[-1]](%10)
  %12 : Float(950, 950, 1, device=cpu) = onnx::Add(%11, %7)
  %13 : Float(950, 950, 1, strides=[950, 1, 1], device=cpu) = onnx::Sqrt(%12)
  %14 : Float(950, 950, 256, strides=[243200, 256, 1], device=cpu) = onnx::Div(%9, %13)
  %15 : Float(950, 950, 256, strides=[243200, 256, 1], device=cpu) = onnx::Mul(%14, %layer.weight) 
  %16 : Float(950, 950, 256, strides=[256, 256, 1], requires_grad=1, device=cpu) = onnx::Add(%15, %layer.bias) 
  return (%16)
```

Note that after `Pow` node, the output shape is `950,950,256` with stride `256,256,1`.

Inference with ONNX Runtime would give warning : `[W:onnxruntime:, graph.cc:119 MergeShapeInfo] Error merging shape info for output. 'output' source:{950,1,256} target:{950,950,256}. Falling back to lenient merge.`
And some fatal error on large model.

![飞书20220906-204407](https://user-images.githubusercontent.com/1239736/188638365-92c9d46d-7702-43b8-893d-fb24d9f87dd1.png)

### Versions

```
'Collecting environment information...
PyTorch version: 1.10.1+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.21.1
Libc version: glibc-2.10

Python version: 3.7.4 (default, Aug 13 2019, 20:35:49)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-4.15.0-189-generic-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: 11.3.109
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1660 SUPER
Nvidia driver version: 510.85.02
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.20.0
[pip3] numpydoc==0.9.1
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.10.1+cu113
[pip3] torch2trt-dynamic==0.5.0
[pip3] torchaudio==0.10.1+rocm4.1
[pip3] torchvision==0.11.2+cu113
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py37h7f8727e_0  
[conda] mkl_fft                   1.3.1            py37hd3c417c_0  
[conda] mkl_random                1.2.2            py37h51133e4_0  
[conda] numpy                     1.20.0                   pypi_0    pypi
[conda] numpydoc                  0.9.1                      py_0  
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torch                     1.10.1+cu113             pypi_0    pypi
[conda] torch2trt-dynamic         0.5.0                     dev_0    <develop>
[conda] torchaudio                0.10.1+rocm4.1           pypi_0    pypi
[conda] torchvision               0.11.2+cu113             pypi_0    pypi

```

**Comment by AllenTiTaiWang** [comment]
Hi @grimoire,

I don't repro your error with the given code and nightly torch.
Do you mind trying with Pytorch nightly?

```bash
onnx graph:  graph(%input : Float(950, 1, 256, strides=[256, 256, 1], requires_grad=0, device=cpu),
      %weight : Float(256, strides=[1], requires_grad=1, device=cpu),
      %bias : Float(256, strides=[1], requires_grad=1, device=cpu)):
  %6 : Double(device=cpu) = onnx::Constant[value={2}](), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %7 : Double(device=cpu) = onnx::Constant[value={1e-05}](), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %8 : Float(950, 1, 1, strides=[1, 1, 1], device=cpu) = onnx::ReduceMean[axes=[-1]](%input), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %9 : Float(950, 1, 256, strides=[256, 256, 1], device=cpu) = onnx::Sub(%input, %8), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %10 : Float(950, 1, 256, strides=[256, 256, 1], device=cpu) = onnx::Pow(%9, %6), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %11 : Float(950, 1, 1, strides=[1, 1, 1], device=cpu) = onnx::ReduceMean[axes=[-1]](%10), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %12 : Float(950, 1, 1, strides=[1, 1, 1], device=cpu) = onnx::Add(%11, %7), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %13 : Float(950, 1, 1, strides=[1, 1, 1], device=cpu) = onnx::Sqrt(%12), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %14 : Float(950, 1, 256, strides=[256, 256, 1], device=cpu) = onnx::Div(%9, %13), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %15 : Float(950, 1, 256, strides=[256, 256, 1], device=cpu) = onnx::Mul(%14, %weight), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  %16 : Float(950, 1, 256, strides=[256, 256, 1], requires_grad=1, device=cpu) = onnx::Add(%15, %bias), scope: torch.nn.modules.normalization.LayerNorm:: # /home/titaiwang/pytorch/torch/nn/functional.py:2511:0
  return (%16)
```

**Comment by grimoire** [comment]
Yes, nightly version did solve the problem! cool!
